dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/ExcIEEEST4B.mo

```diff
@@ -0,0 +1,172 @@
+  parameter Types.PerUnit Kp "Potential circuit gain coefficient.  Typical value = 9,3";
+  parameter Types.PerUnit Kpm "Voltage regulator proportional gain output.  Typical value = 1";
+  parameter Types.PerUnit Kpr "Voltage regulator proportional gain.  Typical value = 10,75";
+  parameter Types.Time tA "Voltage regulator time constant (> 0).  Typical value = 0,02";
+  parameter Types.Angle Thetap "Potential circuit phase angle.  Typical value = 0";
+  parameter Types.Time tR "Filter time constant";
+  parameter Types.VoltageModulePu Ub0Pu "Initial voltage in pu (base UNom)";
+  parameter Types.VoltageModulePu UOel0Pu "Overexciter voltage input in pu (base UNom)";
+  parameter Types.VoltageModulePu VbMaxPu "Maximum excitation voltage (> 0).  Typical value = 11,63";
+  parameter Types.VoltageModulePu VmMaxPu "Maximum inner loop output (> VmMin).  Typical value = 99";
+  parameter Types.VoltageModulePu VmMinPu "Minimum inner loop output (< VmMax).  Typical value = -99";
+  parameter Types.VoltageModulePu VrMaxPu "Maximum voltage regulator output (> 0).  Typical value = 1";
+  parameter Types.VoltageModulePu VrMinPu "Minimum voltage regulator output (< 0).  Typical value = -0,87";
+  parameter Types.PerUnit Xl "Reactance associated with potential source (>= 0).  Typical value = 0,124";
+
+  //Input variables
+  Modelica.Blocks.Interfaces.RealInput IfdPu(start = Ifd0Pu) "Current in pu (base UNom)" annotation(
+    Placement(visible = true, transformation(origin = {-160, -120}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {60, -120}, extent = {{-20, -20}, {20, 20}}, rotation = 90)));
+  Modelica.Blocks.Interfaces.RealInput itPuIm(start = it0Pu.im) annotation(
+    Placement(visible = true, transformation(origin = {-286, -160}, extent = {{-20, 20}, {20, -20}}, rotation = 0), iconTransformation(origin = {-40, -120}, extent = {{-20, -20}, {20, 20}}, rotation = 90)));
+  Modelica.Blocks.Interfaces.RealInput itPuRe(start = it0Pu.re) annotation(
+    Placement(visible = true, transformation(origin = {-286, -140}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-80, -120}, extent = {{-20, -20}, {20, 20}}, rotation = 90)));
+  Modelica.Blocks.Interfaces.RealInput UsPu(start = Us0Pu) annotation(
+    Placement(visible = true, transformation(origin = {-286, 0}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-120, 0}, extent = {{-20, -20}, {20, 20}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealInput UpssPu(start = 0) "PowerSystemStabilizers output voltage in pu (base UNom)" annotation(
+    Placement(visible = true, transformation(origin = {-286, -80}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-120, -60}, extent = {{-20, -20}, {20, 20}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealInput URefPu(start = Us0Pu) "Control voltage in pu (base UNom)" annotation(
```

Comment:
Since it starts at Us0Pu, could it be renamed UsRefPu ?

Reference stator voltage ?

Reply:
I let @cossartque decide.

Reply:
Let's choose to be consistent with other VR